gnucash/import-export/aqb/gnc-flicker-gui.c

```diff
@@ -0,0 +1,408 @@
+        flickerdraw.change_interval = FALSE;
+        return FALSE;
+    }
+    gtk_widget_queue_draw (widget);
+
+    return TRUE;
+}
+
+/* Show the colored triangle as a pointer for the position of the TAN generator */
+void
+doMarkerDrawing (cairo_t *cr)
+{
+	    guint pos1;
+	    guint pos2;
+
+	    /* Initialize the drawing area to black */
+	    cairo_set_source_rgb (cr, 0, 0, 0);
+	    cairo_paint (cr);
+
+	    cairo_set_source_rgb (cr, 0.9, 0.1, 0.1);
+	    /* Dreieck links */
+	    pos1 = flickerdraw.x_drawpos+flickerdraw.barwidth/2;
+	    cairo_move_to(cr, pos1, 20);
+	    cairo_line_to(cr, pos1+10, 2);
+	    cairo_line_to(cr, pos1-10, 2);
+	    cairo_close_path(cr);
+	    cairo_stroke_preserve(cr);
+	    cairo_fill(cr);
+
+	    /* Dreieck rechts */
+	    pos2 = flickerdraw.x_drawpos+4*flickerdraw.margin+4*flickerdraw.barwidth+flickerdraw.barwidth/2;
+	    cairo_move_to(cr, pos2, 20);
+	    cairo_line_to(cr, pos2+10, 2);
+	    cairo_line_to(cr, pos2-10, 2);
+	    cairo_close_path(cr);
+	    cairo_stroke_preserve(cr);
+	    cairo_fill(cr);
+}
+
+/* display the flicker graphic in the drawing area */
+void
+doFlickerDrawing (GtkWidget *widget, cairo_t *cr)
+{
+	guint i;
+
+    /* Initialize the drawing area to black */
+    cairo_set_source_rgb (cr, 0, 0, 0);
+    cairo_paint (cr);
+
+    bitarray[flickerdraw.halfbyteid][0] = flickerdraw.clock;
+
+    /* die Flickeranzeige immer in der Mitte der Drawingarea ausrichten */
+    flickerdraw.width = gtk_widget_get_allocated_width (widget);
+    flickerdraw.x_drawpos = (flickerdraw.width - 4*flickerdraw.margin - 5*flickerdraw.barwidth)/2;		/* Startposition des ersten Balkens */
+
+    for ( i = 0; i <= 4; i++ ) {
+    	if ( bitarray[flickerdraw.halfbyteid][i] == TRUE ) {
+    		cairo_set_source_rgb(cr, 1, 1, 1); // white
+    	}
+    	else {
+    		cairo_set_source_rgb(cr, 0, 0, 0); // black
+    	}
+    	flickerdraw.x_barpos = flickerdraw.x_drawpos + i*flickerdraw.margin + i*flickerdraw.barwidth;
+    	cairo_rectangle (cr, flickerdraw.x_barpos, flickerdraw.y_barpos, flickerdraw.barwidth, flickerdraw.barheight);
+    	cairo_fill (cr);
+    }
+
+    /* Each flicker point is drawn twice. Once with clock = 1 and once with clock = 0 */
+    if ( flickerdraw.clock == 0 ) {
+        flickerdraw.clock = 1;
+        flickerdraw.halfbyteid++;
+        if ( flickerdraw.halfbyteid >= flickerdraw.maxWert ) {
+            flickerdraw.halfbyteid = 0;
+        }
+    }
+    else if ( flickerdraw.clock == 1 ) {
+        flickerdraw.clock = 0;
+    }
+}
+
+static void
+doFlickerLoadState (void)
+{
+    /* Fenstergröße und Einstellwerte laden */
+    GSettings *settings;
+
+    settings = g_settings_new ("org.gnucash.dialogs.flicker");
+    g_settings_get (settings, "window-size", "(ii)", &flickerdraw.win_width, &flickerdraw.win_height);
+    flickerdraw.barwidth = g_settings_get_uint (settings, "barwidth");
+    flickerdraw.delay = g_settings_get_uint (settings, "delay");
+    g_object_unref (settings);
```

Comment:
For saving to state file change this to...
    GKeyFile *state_file = gnc_state_get_current ();

     if (g_key_file_has_key (state_file, GNC_STATE_SECTION, STATE_KEY_BARWIDTH, NULL))
          flickerdraw.barwidth = g_key_file_get_integer (state_file, GNC_STATE_SECTION, STATE_KEY_BARWIDTH, NULL);
     else
         flickerdraw.barwidth = BARWIDTH;

     if (g_key_file_has_key (state_file, GNC_STATE_SECTION, STATE_KEY_DELAY, NULL))
          flickerdraw.barwidth = g_key_file_get_integer (state_file, GNC_STATE_SECTION, STATE_KEY_DELAY, NULL);
     else
          flickerdraw.delay = DELAY;

     gnc_restore_window_size (GNC_PREFS_GROUP,
                              GTK_WINDOW(flickergui->dialog), gnc_ui_get_main_window (NULL));

Reply:
@Bob-IT: it seems that gnc_restore_window_size () and gnc_save_window_size () are not working. However, there is no error message either, debugging looks good.